### PR TITLE
Add fips self test DEP for solaris and hpux

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by mattcaswell** [comment]
Travis red cross is not relevant.

**Comment by levitte** [comment]
> Travis red cross is not relevant.

That build job consistently fails with a timeout. We need to take it off, or deal with it. 
Not for this PR, though

File: providers/fips/selftest.c [hunk 1]
```diff
@@ -85,14 +85,44 @@ BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
...
+void dep_init(void)
+{
+    FIPS_state = FIPS_STATE_SELFTEST;
+}
+
+#pragma fini "dep_cleanup"
+void dep_cleanup(void)
+{
+    CRYPTO_THREAD_lock_free(self_test_lock);
+}
```
Comment (mattcaswell) [inline providers/fips/selftest.c]: Could we use init/cleanup names which are consistent with the GNUC version?

Also I wonder if we could somehow combine all these function implementations together somehow. They're all basically the same function implementations with variations around how we declare them or mark them as init/fini functions.

Comment (slontis) [reply providers/fips/selftest.c]: Ok I have attempted to do the second thing..
you cant put #pragmas inside macros so it does become a bit messy...

File: providers/fips/selftest.c [hunk 2]
```diff
@@ -50,6 +50,10 @@ DEFINE_RUN_ONCE_STATIC(do_fips_self_test_init)
     return self_test_lock != NULL;
 }
 
+#define DEP_DELCARE()                                                          \
```
Comment (levitte) [inline providers/fips/selftest.c]: Who's care? 😉

Comment (slontis) [reply providers/fips/selftest.c]: Not me :) Least I spelt it consistently wrong..
Updated

**Comment by slontis** [comment]
Merged to master.

**Pull request closed without merging**